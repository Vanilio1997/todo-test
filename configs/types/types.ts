export interface IBuildPaths {
    entry: string;
    html: string;
    output: string;
    src: string;
}

export type BuildMode = 'development' | 'production';
export type BuildPlatform = 'mobiel' | 'desktop';

export interface IBuildOptions {
    port: number;
    paths: IBuildPaths;
    mode: BuildMode;
    analyzer?: boolean;
    platform: BuildPlatform;
}
