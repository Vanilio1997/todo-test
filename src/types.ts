import * as React from 'react';

export interface LayoutProps {
    children: React.ReactNode;
}

export type IconProp = React.ComponentType<React.SVGProps<SVGSVGElement>>;

export interface ISVGIcon {
    icon: IconProp;
    style?: React.CSSProperties;
    color?: string;
}

export interface ITask {
    isCompleted: boolean;
    text: string;
    id: number;
}

export type handleStatusTaskChangeType = (index: number) => void;

export type DeleteTaskType = (type: ITask) => void;

export type ChangeTaskTextType = (index: number, newText: string) => void;

export type FilterType = 'all' | 'active' | 'completed';
