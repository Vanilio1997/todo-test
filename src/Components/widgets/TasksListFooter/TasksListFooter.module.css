.footer {
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding: 16px;
    font-size: 14px;
    color: #777;
    border-top: 1px solid #e6e6e6;
}
@media (max-width: 720px) {
    .footer {
        flex-direction: column;
        gap: 8px;
        text-align: center;
    }
}

.itemsLeft {
    flex: 1;
}

.tabsContainer {
    position: absolute;
    left: 50%;
    transform: translateX(-50%);
}

.clearButton {
    flex: 1;
    text-align: right;
    background: none;
    border: none;
    color: #777;
    cursor: pointer;
    padding: 3px 7px;
    font-size: 14px;
}

.clearButton:hover {
    text-decoration: underline;
}
