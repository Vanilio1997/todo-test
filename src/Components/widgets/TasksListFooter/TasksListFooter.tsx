import Tabs from '@/Components/ui/Tabs/Tabs';
import { FilterType } from '@/types';
import * as styles from './TasksListFooter.module.css';

interface ITasksListFooter {
    itemsLeft: number;
    activeTab: FilterType;
    changeFilterType: (type: string) => void;
    clearCompleted: () => void;
}

const tabsValues: string[] = ['all', 'active', 'completed'];

const TasksListFooter = ({
    activeTab,
    changeFilterType,
    clearCompleted,
    itemsLeft,
}: ITasksListFooter) => {
    return (
        <div className={styles.footer}>
            <div className={styles.itemsLeft}>{itemsLeft} items left</div>
            <Tabs
                tabs={tabsValues}
                activeTab={activeTab}
                changeActiveTab={changeFilterType}
            />
            <button className={styles.clearButton} onClick={clearCompleted}>
                Clear completed
            </button>
        </div>
    );
};

export default TasksListFooter;
