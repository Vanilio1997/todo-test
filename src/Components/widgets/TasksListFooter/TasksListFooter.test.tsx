import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import TasksListFooter from './TasksListFooter';
import { FilterType } from '@/types';

const mockProps = {
    itemsLeft: 3,
    activeTab: 'all' as FilterType,
    changeFilterType: jest.fn(),
    clearCompleted: jest.fn(),
};

describe('TasksListFooter Component', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    test('renders correct items left count', () => {
        render(<TasksListFooter {...mockProps} />);
        expect(screen.getByText('3 items left')).toBeInTheDocument();
    });

    test('renders all filter tabs', () => {
        render(<TasksListFooter {...mockProps} />);
        expect(screen.getByRole('tab', { name: 'all' })).toBeInTheDocument();
        expect(screen.getByRole('tab', { name: 'active' })).toBeInTheDocument();
        expect(
            screen.getByRole('tab', { name: 'completed' }),
        ).toBeInTheDocument();
    });

    test('highlights active tab', () => {
        render(<TasksListFooter {...mockProps} activeTab="active" />);
        const activeTab = screen.getByRole('tab', { name: 'active' });
        expect(activeTab).toHaveAttribute('aria-selected', 'true');
    });

    test('calls changeFilterType when tab clicked', () => {
        render(<TasksListFooter {...mockProps} />);
        const completedTab = screen.getByRole('tab', { name: 'completed' });
        fireEvent.click(completedTab);
        expect(mockProps.changeFilterType).toHaveBeenCalledWith('completed');
    });

    test('renders clear completed button', () => {
        render(<TasksListFooter {...mockProps} />);
        expect(
            screen.getByRole('button', { name: 'Clear completed' }),
        ).toBeInTheDocument();
    });

    test('calls clearCompleted when button clicked', () => {
        render(<TasksListFooter {...mockProps} />);
        const clearButton = screen.getByRole('button', {
            name: 'Clear completed',
        });
        fireEvent.click(clearButton);
        expect(mockProps.clearCompleted).toHaveBeenCalled();
    });

    test('does not render clear button when no completed tasks', () => {
        render(<TasksListFooter {...mockProps} itemsLeft={0} />);
        expect(
            screen.queryByRole('button', { name: 'Clear completed' }),
        ).toBeInTheDocument();
    });

    test('matches snapshot', () => {
        const { container } = render(<TasksListFooter {...mockProps} />);
        expect(container).toMatchSnapshot();
    });
});
