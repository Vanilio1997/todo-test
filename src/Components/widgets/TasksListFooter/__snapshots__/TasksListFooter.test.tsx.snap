// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`TasksListFooter Component matches snapshot 1`] = `
<div>
  <div>
    <div>
      3
       items left
    </div>
    <div
      role="tablist"
    >
      <div>
        <button
          aria-controls="tabpanel-0"
          aria-selected="true"
          class="undefined undefined"
          id="tab-0"
          role="tab"
          tabindex="0"
        >
          all
        </button>
        <button
          aria-controls="tabpanel-1"
          aria-selected="false"
          class="undefined "
          id="tab-1"
          role="tab"
          tabindex="-1"
        >
          active
        </button>
        <button
          aria-controls="tabpanel-2"
          aria-selected="false"
          class="undefined "
          id="tab-2"
          role="tab"
          tabindex="-1"
        >
          completed
        </button>
      </div>
    </div>
    <button>
      Clear completed
    </button>
  </div>
</div>
`;
